lib/server.py

```diff
@@ -88,12 +92,19 @@ def __init__(self, ini, test_suite=None):
     def prepare_args(self, args=[]):
         return args
 
-    def cleanup(self, full=False):
+    def pre_cleanup(self):
+        self.cleanup()
+
+    def cleanup(self, full=False, dirname='.', patterns=DEFAULT_CHECKPOINT_PATTERNS):
```

Comment:
`full` is not used, proposed to remove.

Reply:
Other parameters are also don't used. Proposed to remove too.

I'm against introducing extra complexity until it is needed.

Reply:
fixed